README.md

```diff
@@ -27,16 +36,72 @@ represents the "RGB vector space" version.
 
 ![ColorVectorSpace](images/comparison.png "Comparison")
 
+This package also defines `norm(c)` for RGB and grayscale colors.
+This makes these color spaces [normed vector spaces](https://en.wikipedia.org/wiki/Normed_vector_space).
+Note that `norm` has been designed to satisfy equivalence of grayscale and RGB representations: if
+`x` is a scalar, then `norm(x) == norm(Gray(x)) == norm(RGB(x, x, x))`.
+
 ## Usage
 
 ```jl
 using ColorTypes, ColorVectorSpace
 ```
 
-That's it. Just by loading `ColorVectorSpace`, most basic mathematical
+For the most part, that's it; just by loading `ColorVectorSpace`, most basic mathematical
 operations will "just work" on `AbstractRGB`, `AbstractGray`
 (`OpaqueColor{T,1}`), `TransparentRGB`, and `TransparentGray` objects.
 (See definitions for the latter inside of `ColorTypes`).
 
-If you discover missing operations, please open an issue, or better
-yet submit a pull request.
+However, there are some additional operations.
+
+### Multiplication
+
+Grayscale values are conceptually similar to scalars, and consequently it seems straightforward to define multiplication of two grayscale values.
+RGB values present more options.
+This package supports three different notions of multiplication: the inner product, the hadamard (elementwise) product, and the tensor product.
+
+```julia
+julia> c1, c2 = RGB(0.2, 0.3, 0.4), RGB(0.5, 0.3, 0.2)
+(RGB{Float64}(0.2,0.3,0.4), RGB{Float64}(0.5,0.3,0.1))
+
+julia> c1⋅c2     # or dot(c1, c2)
+0.09000000000000001
+
+julia> c1⊙c2     # or hadamard(c1, c2)
+RGB{Float64}(0.1,0.09,0.08000000000000002)
+
+julia> c1⊗c2     # or tensor(c1, c2)
+RGBRGB{Float64}(
+ 0.1   0.06  0.04000000000000001
+ 0.15  0.09  0.06
+ 0.2   0.12  0.08000000000000002)
+```
+
+Note that `c1⋅c2 = (c1.r*c2.r + c1.g*c2.g + c1.b*c2.b)/3`, where the division by 3 ensures the equivalence of `x`, `Gray(x)`, and `RGB(x, x, x)` for scalar `x` as described above for `norm`.
```

Comment:
I don't really like this definition. I don't think `⋅` needs to return the "consistent" values with `norm`, from a point of view in programming, not math. Giving different definitions different names (symbols) is exactly what this PR is trying to do.

I also don't really like the definition of `norm`. As you know, the definition of  norm affects the definition of unit vector.
It's a bit strange to equate `Gray(x)` with `RGB(x, x, x)`. More obvious inconsistencies can be found in `TransparentColor`.
```julia
julia> norm(Gray(0.5)) == norm(AGray(0.5))
false
```
We apply the promotion rules(implicit type conversions) on the addition, so it doesn't "seem" to satisfy the triangle inequality. (Of course, it is satisfied in the post-promotion vector space.)

Reply:
We should probably get rid of `norm(::TransparentColor)`.

If you're concerned about the interpretation of `dot`/`⋅` as the Euclidean dot product of a literal vector `[r,g,b]`, then we should change the name/symbol of the operator, not this definition. I am planning on enforcing strict equivalence throughout, it is vital and one of the only guiding principles available.

However, personally I don't think it's an issue. For example we could define
```julia
echannels(g::Gray) = (gray(g),)
echannels(c::RGB) = (red(c)/SQRT3, green(c)/SQRT3, blue(c)/SQRT3)
```
and then no one would argue with the thought that this definition is equivalent to the Euclidean dot product of `echannels(c1) ⋅ echannels(c2)`. In other words, this is just a decision about how you embed these otherwise-opaque objects into a vector space.